cpp/src/io/functions.cpp

```diff
@@ -149,6 +149,17 @@ table_with_metadata read_csv(read_csv_args const& args, rmm::mr::device_memory_r
   }
 }
 
+// Freeform API wraps the detail writer class API
+void write_csv(write_csv_args const& args,
+               rmm::mr::device_memory_resource* mr) {
+  using namespace cudf::experimental::io::detail;
+  
+  csv::writer_options options{args.get_options()};
+  auto writer = make_writer<csv::writer>(args.sink(), options, mr);
```

Comment:
Now that write_csv_args inherits from writer_options, you should be able to just pass args where a writer_option& is needed, eliminating the need for the local `options`.

Reply:
Done.